Update to use MSAL 0.4.0

jasoncoolmax opened the pull request.

jasoncoolmax committed:
Improve status message

jasoncoolmax committed:
Update to MSAL 0.4.0

jasoncoolmax committed:
update cart file

jasoncoolmax committed:
Remove unnecessary code

jasoncoolmax committed:
Change silent call and add some comments

jasoncoolmax committed:
Remove unneccessary init

antrix1989 commented on O365-iOS-Microsoft-Graph-Connect-Swift/AuthenticationClass.swift:
```diff
@@ -10,7 +10,7 @@ class AuthenticationClass {
     static let sharedInstance = AuthenticationClass()
 
-    var authenticationProvider = MSALPublicClientApplication.init()
+    var authenticationProvider = MSALPublicClientApplication()
```
Doing this way we don't need to init empty MSALPublicClientApplication:

`var authenticationProvider: MSALPublicClientApplication!`

jasoncoolmax replied:
agree, fixed!

antrix1989 commented on O365-iOS-Microsoft-Graph-Connect-Swift/AuthenticationClass.swift:
```diff
@@ -50,27 +53,29 @@ class AuthenticationClass {
-            guard !(try authenticationProvider.users().isEmpty) else {
-                throw NSError.init(domain: "MSALErrorDomain",
-                                   code: MSALErrorCode.interactionRequired.rawValue,
+            // Acquire a token for an existing user silently
```
Should we show here how to handle other errors (for example 'scopes declined' error)? See an example [here](https://github.com/microsoftgraph/ios-swift-snippets-sample/pull/13/files#diff-5d69b93abbd01a3e15b20579c104fbc4R24).

antrix1989 commented on O365-iOS-Microsoft-Graph-Connect-Swift/AuthenticationClass.swift:
```diff
@@ -97,7 +103,12 @@ class AuthenticationClass {
 
     func disconnect() {
-        try? authenticationProvider.remove(authenticationProvider.users().first)
+        let accounts = try? authenticationProvider.allAccounts()
```
It can be simplified to:
 ```
let accounts = try? authenticationProvider.allAccounts()
guard let account = accounts?.first else { return }

try? authenticationProvider.remove(account)
```

jasoncoolmax replied:
:+1: fixed

antrix1989 commented on O365-iOS-Microsoft-Graph-Connect-Swift/O365-iOS-Microsoft-Graph-Connect-Swift.entitlements:
```diff
@@ -5,6 +5,7 @@
 	<key>keychain-access-groups</key>
 	<array>
 		<string>$(AppIdentifierPrefix)com.microsoft.O365-iOS-Microsoft-Graph-Connect-Swift-REST</string>
+		<string>$(AppIdentifierPrefix)com.microsoft.adalcache</string>
```
👍

antrix1989 commented:
Should we use only one dependency management? I think we can drop Carthage and use Cocoapod for MSAL, or we can move PromiseKit to Carthage and drop Cocoapod file.

jasoncoolmax committed:
Improve code

oldalton commented on O365-iOS-Microsoft-Graph-Connect-Swift/AuthenticationClass.swift:
```diff
@@ -30,12 +30,15 @@ class AuthenticationClass {
         } catch let error as NSError {
             self.lastInitError = error.userInfo.description
-            authenticationProvider = MSALPublicClientApplication.init()
+            authenticationProvider = MSALPublicClientApplication()
```
why are we initializing an empty one?

oldalton reviewed (commented):
Can you please also add msauth and msauthv2 query schemes into Info.plist, so that broker could be used?

oldalton commented on O365-iOS-Microsoft-Graph-Connect-Swift/AuthenticationClass.swift:
```diff
@@ -30,12 +30,15 @@ class AuthenticationClass {
             let kClientId = String(redirectUrl[msalRange.upperBound...])
+            let authority = try MSALAADAuthority(url: URL(string:ApplicationConstants.kAuthority)!)
+            
+            // Redirect uri will be constucted automatically in the form of "msal<your-client-id-here>://auth" if not provided.
```
Let's update to msauth.bundleId format so that broker could be used

jasoncoolmax replied:
broker support added :)

oldalton commented on O365-iOS-Microsoft-Graph-Connect-Swift/AuthenticationClass.swift:
```diff
@@ -50,27 +53,29 @@ class AuthenticationClass {
         do {
             if let initError = self.lastInitError {
-                throw NSError.init(domain: initError, code: 0, userInfo: nil)
+                throw NSError(domain: initError, code: 0, userInfo: nil)
```
why is this one necessary? Specifically with code:0?

jasoncoolmax replied:
I didn't change much on the original logic. But here seems very weird. I have changed to throw the init error rather than creating a new error here.

jasoncoolmax committed:
broker support

jasoncoolmax committed:
Update readme

jasonjoh committed:
Merge branch 'master' into jason/updateMSAL

jasonjoh committed:
Switched install of MSAL to CocoaPods

Project was already using CocoaPods for another dependency, and the Carthage install kept timing out.
